ee/server/service/mdm.go

```diff
@@ -914,6 +915,29 @@ func (svc *Service) getOrCreatePreassignTeam(ctx context.Context, groups []strin
 		if err := svc.ds.CopyDefaultMDMAppleBootstrapPackage(ctx, ac, team.ID); err != nil {
 			return nil, err
 		}
+
+		// get the global setup assistant contents (this is different
+		// from MDM.MacOSSetup.MacOSSetupAssistant we set above, the
+		// prior is the path to the file, this is the actual file
+		// contents.
+		asst, err := svc.GetMDMAppleSetupAssistant(ctx, nil)
```

Comment:
Using the service means that it will require a different set of authorization (i.e. `MDMAppleMatchPreassignment` - which calls `getOrCreatePreassignTeam` - requires write access to `mdm_apple_config_profile`, while `GetMDMAppleSetupAssistant` requires read access to `mdm_apple_setup_assistant`).

I think it'd be best to use a direct datastore access to retrieve the apple setup assistant (`svc.ds.GetMDMAppleSetupAssistant`) so that the Apple match preassignment endpoint does not require extra authorization (it's an implementation detail that it needs to read the existing global setup assistant, not something that the caller would see, so a "write-only" access like `gitops` could be possible).

Reply:
(and thanks for the detailed explanation)